square.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+type Point struct {
+	x, y int
+}
+
+type Square struct {
+	start Point
+	a     uint
+}
+
+func (s Square) End() Point {
+	/* some hardcoded stuff
+	i used the minus sign in calculation of y axis because in the attached screenshot start.y lower then end.y (???)
+	*/
+	return Point{s.start.x + int(s.a), s.start.y - int(s.a)}
+}
+
+func (s Square) Perimeter() uint {
+	const squareEdgesCount = 4
```

Comment:
You can use just magic number there as a contact it obvious what "4" means in scope of this function